cpp/src/groupby/hash/groupby.cu

```diff
@@ -84,9 +94,114 @@ bool constexpr is_hash_aggregation(aggregation::Kind t)
   // return array_contains(hash_aggregations, t);
   return (t == aggregation::SUM) or (t == aggregation::MIN) or (t == aggregation::MAX) or
          (t == aggregation::COUNT_VALID) or (t == aggregation::COUNT_ALL) or
-         (t == aggregation::ARGMIN) or (t == aggregation::ARGMAX);
+         (t == aggregation::ARGMIN) or (t == aggregation::ARGMAX) or (t == aggregation::MEAN) or
+         (t == aggregation::SUM_OF_SQUARES) or (t == aggregation::STD) or
+         (t == aggregation::VARIANCE);
 }
 
+class hash_compound_agg_finalizer final : public cudf::detail::aggregation_finalizer {
+  size_t i;
+  column_view col;
+  cudf::detail::result_cache& sparse_results;
+  cudf::detail::result_cache* dense_results;
+  rmm::device_vector<size_type> const& gather_map;
+  size_type const map_size;
+  rmm::mr::device_memory_resource* mr;
+  cudaStream_t stream;
+
+ public:
+  hash_compound_agg_finalizer(size_t col_idx,
+                              column_view col,
+                              cudf::detail::result_cache& sparse_results,
+                              cudf::detail::result_cache* dense_results,
+                              rmm::device_vector<size_type> const& gather_map,
+                              size_type map_size,
+                              cudaStream_t stream,
+                              rmm::mr::device_memory_resource* mr)
+    : i(col_idx),
+      col(col),
+      sparse_results(sparse_results),
+      dense_results(dense_results),
+      gather_map(gather_map),
+      map_size(map_size),
+      stream(stream),
+      mr(mr)
+  {
+  }
+
+  auto to_dense_agg_result(cudf::aggregation const& agg)
+  {
+    auto s                  = sparse_results.get_result(i, agg);
+    auto dense_result_table = cudf::detail::gather(
+      table_view({s}), gather_map.begin(), gather_map.begin() + map_size, false, mr, stream);
+    return std::move(dense_result_table->release()[0]);
+  }
+
+  // Enables conversion of ARGMIN/ARGMAX into MIN/MAX
+  auto transformed_result(aggregation::Kind const& agg_kind)
+  {
+    auto transformed_agg = std::make_unique<aggregation>(agg_kind);
+    auto arg_result      = to_dense_agg_result(*transformed_agg);
+    // We make a view of ARG(MIN/MAX) result without a null mask and gather
+    // using this map. The values in data buffer of ARG(MIN/MAX) result
+    // corresponding to null values was initialized to ARG(MIN/MAX)_SENTINEL
+    // which is an out of bounds index value (-1) and causes the gathered
+    // value to be null.
```

Comment:
Why does `arg_result` have a null mask in the first place?

Reply:
because ARG_MIN, ARG_MAX sparse results contains null mask. 
All-null group could cause ARG_MIN/ARG_MAX to contain null.
Converting it to dense results using gather may contain nulls.